metricbeat/module/nginx/stubstatus/data.go

```diff
@@ -0,0 +1,101 @@
+package stubstatus
+
+import (
+	"bufio"
+	"io"
+	"regexp"
+	"strconv"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// Map body to MapStr
+func eventMapping(m *MetricSeter, body io.ReadCloser, hostname string, metricset string) common.MapStr {
+	// Nginx stub status sample:
+	// Active connections: 1
+	// server accepts handled requests
+	//  7 7 19
+	// Reading: 0 Writing: 1 Waiting: 0
+	var (
+		active   int
+		accepts  int
+		handled  int
+		dropped  int
+		requests int
+		current  int
+		reading  int
+		writing  int
+		waiting  int
+	)
+
+	var re *regexp.Regexp
+	scanner := bufio.NewScanner(body)
+
+	// Parse active connections.
+	scanner.Scan()
+	re = regexp.MustCompile("Active connections: (\\d+)")
+	if matches := re.FindStringSubmatch(scanner.Text()); matches == nil {
+		logp.Warn("Fail to parse active connections from Nginx stub status")
+		active = -1
+	} else {
+		active, _ = strconv.Atoi(matches[1])
+	}
+
+	// Skip request status headers.
+	scanner.Scan()
+
+	// Parse request status.
+	scanner.Scan()
+	re = regexp.MustCompile("\\s(\\d+)\\s+(\\d+)\\s+(\\d+)")
+	if matches := re.FindStringSubmatch(scanner.Text()); matches == nil {
+		logp.Warn("Fail to parse request status from Nginx stub status")
+		accepts = -1
+		handled = -1
+		dropped = -1
+		requests = -1
+		current = -1
+	} else {
+		accepts, _ = strconv.Atoi(matches[1])
```

Comment:
Hm, good point about the apache module. I have to think about it again how we should handle this cases in general. Somehow it would be nice if 1 missing field would not crash the hole metricset, but that we are still notified about the error(s). So perhaps an error array should be returned and the available data.

I suggest to keep here as it is for the moment and do a general refactoring on all modules as soon as we have it in.